File: Model.Tests/TournamentTests.vb
<TestClass()> Public Class TournamentTests

    <TestMethod()> Public Sub TournamentTest()
        ' Arrange
        ' -

        ' Act
        Dim result = New Tournament()

        ' Assert
        Assert.IsNotNull(result.TeamManager, "TeamManager is null")
        Assert.IsNotNull(result.MatchManager, "MatchManager is null")
    End Sub

    <TestMethod()> Public Sub GetStandingsTest()
        ' Arrange
        Dim testClass As Tournament = CreateTournementWith2Teams()

        ' Act
        Dim result = testClass.GetStandings()

        ' Assert
        Assert.IsNotNull(result, "GetStandings returned null")
        Assert.AreEqual(2, result.Count, "Unexpected Count")
    End Sub

    Private Shared Function CreateTournementWith2Teams() As Tournament
        Dim result = New Tournament()
        Dim team1 = New Team(1)
        Dim team2 = New Team(2)
        result.AddTeam(team1)
        result.AddTeam(team2)
        result.MatchManager.Generate({team1, team2})
        Dim matchResult = New Result()
        matchResult.AddSetResult(New SetResult() With {.Home = 5, .Guest = 3})
        result.MatchManager.SetStatus(1, matchResult)
        Return result
    End Function

    <TestMethod()> Public Sub ExportTest()
        Dim testClass = CreateTournementWith2Teams()
        testClass.Export()
    End Sub

End Class

File: POFF.Kicker.Tests/TournamentTests.cs
using System.Linq;
using NUnit.Framework;
using POFF.Kicker.Domain;
using POFF.Kicker.View.Model;

namespace POFF.Kicker.Tests;

[TestFixture]
class TournamentTests
{
    [Test]
    public void TournamentTest()
    {
        // Arrange
        // -

        // Act
        var result = new Tournament();

        // Assert
        Assert.That(result.Teams, Is.Not.Null);
        Assert.That(result.MatchManager, Is.Not.Null);
    }

    [Test]
    public void GetStandingsTest()
    {
        // Arrange
        var testClass = new Tournament();
        var team1 = new Team(1);
        var team2 = new Team(2);
        testClass.AddTeam(team1);
        testClass.AddTeam(team2);
        testClass.Start(TournamentType.Standard);
        var matchResult = new Result();
        matchResult.AddSetResult(new SetResult() { Home = 5, Guest = 3 });
        testClass.MatchManager.SetStatus(1, matchResult);

        // Act
        var result = testClass.GetStandings();

        // Assert
        Assert.That(result, Is.Not.Null, "GetStandings returned null");
        Assert.That(result.Count(), Is.EqualTo(2), "Unexpected Count");
    }
}